src/houghtransform.jl

```diff
@@ -0,0 +1,102 @@
+
+#function to compute local maximum lines with values > threshold and return a vector containing them
+function findlocalmaxima(accumulator_matrix::Array{Integer,2},threshold::Integer)
+    validLines = Vector{CartesianIndex}(0)
+    for val in CartesianRange(size(accumulator_matrix))
+        if  accumulator_matrix[val] > threshold &&
+            accumulator_matrix[val] > accumulator_matrix[val[1],val[2] - 1] &&
+            accumulator_matrix[val] >= accumulator_matrix[val[1],val[2] + 1] &&
+            accumulator_matrix[val] > accumulator_matrix[val[1] - 1,val[2]] &&
+            accumulator_matrix[val] >= accumulator_matrix[val[1] + 1,val[2]]
+            push!(validLines,val)
+        end
+    end
+    validLines
+end
+
+"""
+```
+lines = hough_transform_standard(image, rho, theta, min_theta, max_theta, threshold, linesMax)
+```
+
+Returns an vector of tuples corresponding to the tuples of (r,t) where r and t are parameters for normal form of line:
+    x*cos(t) + y*sin(t) = r
+
+r = length of perpendicular from (0,0) to the line
+t = angle between perpendicular from (0,0) to the line and axis
+
+The lines are generated by applying hough transform on the image.
+
+Parameters:
+    image       = image to apply hough transform on
+    rho         = discrete steps for radius
+    theta       = discrete steps for theta
+    min_theta   = minimum theta of lines
+    max_theta   = maximum theta of lines
+    threshold   = no of points to pass through line for considering it valid
+    linesMax    = maximum no of lines to return
+
+"""
+
+function hough_transform_standard{T<:Union{Bool,Gray{Bool}}}(
```

Comment:
Would it be a nicer interface to pass in a `Range` for `θ`, e.g. `linspace(min_theta, max_theta, theta)`? That reduces the number of arguments needed for this function. You can get `θ` with `\theta TAB` (no space in between).